shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerContainerComposer.java

```diff
@@ -50,8 +54,12 @@ public final class DockerContainerComposer extends BaseContainerComposer {
     public DockerContainerComposer(final DatabaseType databaseType, final String storageContainerImage) {
         this.databaseType = databaseType;
         governanceContainer = getContainers().registerContainer(new ZookeeperContainer());
+        StorageContainerConfiguration storageContainerConfig = StorageContainerConfigurationFactory.newInstance(databaseType);
+        if (DatabaseTypeUtil.isMySQL(databaseType) && new DockerImageVersion(storageContainerImage).getMajorVersion() > 5) {
+            storageContainerConfig = ScalingMySQL8ContainerConfigurationFactory.newInstance();
+        }
```

Comment:
Could we define different `my.cnf` for MySQL 8 and other MySQL version, it might be more flexible to define more dedicated configuration for different MySQL versions. And ScalingMySQL8ContainerConfigurationFactory might be not required.

Reply:
Yes, but the MySQLContainerConfigurationFactory is final and all method is static, can't change the mount file

Reply:
Maybe we need to refactor `MySQLContainerConfigurationFactory` as builder pattern

Reply:
I will think about how to optimize, this pr first delete the new factory